addon/classes/ChangesetArrayProxyHandler.ts

```diff
@@ -0,0 +1,298 @@
+import ArrayProxy from '@ember/array/proxy';
+import { action } from '@ember/object';
+import { A } from '@ember/array';
+import { TrackedMap } from 'tracked-maps-and-sets';
+import IChangesetProxyHandler from './IChangesetProxyHandler';
+import ProxyOptions from './ProxyOptions';
+import { ProxyArrayValueKey } from 'ember-changeset/utils/symbols';
+
+type Input = any[] | ArrayProxy<any>;
+
+export default class ChangesetArrayProxyHandler<T extends Input> implements IChangesetProxyHandler<T> {
+  constructor(source: T, _options?: ProxyOptions) {
+    if (source?.constructor?.name === 'ArrayProxy') {
+      this.__sourceProxy = source as ArrayProxy<any>;
```

Comment:
these should either by symbol properties, or private properties, as they may conflict with folks data

Reply:
you mean __sourceProxy etc?

Reply:
I don't think these do conflict. Proxy is not a wrapper that falls back to the internals like Ember.ObjectProxy. It explicitly calls in with `get` and `set` and you do your own fallback. __sourceProxy is private api so it's not available through get and set, and if there was a name conflict it would fall through to the proxied object correctly.